### PR TITLE
Initial Release

### LINKED ISSUE #2: Support negative values

### PR DESCRIPTION
**Pull request by theiliad**
Closes #2, #5, #7, #8 

## Types of charts included:
1. Simple Bar
2. Grouped Bar
3. Stacked Bar
4. Donut
5. Line
6. Curved Line
7. Pie
8. Step
9. Area

## Summary:
- Data-driven rendering pipeline
- Consistent configuration object schema
- Dataset support
- Most general properties & logic is inherited from BaseChart & BaseAxisChart
- Built-in Animations
- Patterns Service
- Accessibility mode using patterns
- Events (onLoad, onDataChange...)
- Curved line charts (d3 built-in curves can be used http://bl.ocks.org/d3indepth/b6d4845973089bc1012dec1674d3aff8)
- Formatters for all axes
- Legend filter persistence in data changes
- Series & labels based legend

### Review checklist (for reviewers only)
- [ ] Demos all features
- [ ] Documented/annotated
- [ ] Matches UI/UX specs
- [ ] Accessible
- [ ] Mobile first (responsive)
- [ ] RTL support (bidirectional text)
- [ ] Performant (limited bloat)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,9 +6,14 @@ node_js:
 #   chrome: stable
 # before_install:
 # - google-chrome-stable --headless --disable-gpu --remote-debugging-port=9222 http://localhost &
-before_install: .travis/before_install.sh
-after_success: .travis/after_success.sh
-# script: npm test && npm run lint && cd packages/core && npm run demo:build
+before_install:
+  - chmod +x .travis/before_install.sh
```
Comment (zvonimirfras) [inline .travis.yml]: You should only have to do this once on your computer before pushing the file. Flags should copy. 🤞

File: .travis.yml [hunk 2]
```diff
@@ -6,9 +6,14 @@ node_js:
...
+  - .travis/before_install.sh
+after_success:
+  - chmod +x .travis/after_success.sh
```
Comment (zvonimirfras) [inline .travis.yml]: You should only have to do this once on your computer before pushing the file. Flags should copy.

File: packages/core/demo/index.html [hunk 3]
```diff
@@ -17,12 +17,16 @@
...
-				<a class="bx--btn bx--btn--primary" href="https://github.com/IBM/carbon-charts">
-					Project Repository
+				<a class="bx--btn bx--btn--primary" href="https://github.com/IBM/carbon-charts" target="_blank">
+					Repository
 				</a>
 
-				<a class="bx--btn bx--btn--primary" href="https://pages.github.com/IBM/carbon-charts/documentation/">
-					Documentation
+				<a class="bx--btn bx--btn--primary" href="https://charts.carbondesignsystem.com/documentation/" target="_blank">
+					Docs
```
Comment (zvonimirfras) [inline packages/core/demo/index.html]: What's wrong with full words?

Comment (theiliad) [reply packages/core/demo/index.html]: I think docs sufficiently explains where the link is pointing to, also it's the *slang* most developers use for documenatations

**Pull request merged**